support/builder/s3migrate.sh

```diff
@@ -0,0 +1,360 @@
+#!/bin/bash
+set -euo pipefail
+
+aggMsg() {
+    echo ""
+    echo "Aggregating locally stored artifacts."
+    echo "======================================"
+    echo ""
+
+}
+
+configMsg(){
+    echo ""
+    echo "Configuring environment for migration."
+    echo "======================================"
+    echo ""
+}
+
+keyMsg() {
+    echo ""
+    echo "Generating S3 Object Keys."
+    echo "======================================"
+    echo ""
+}
+
+uploadMsg() {
+    echo ""
+    echo "Uploading hartfile."
+    echo "======================================"
+}
+
+getHarts() {
+    # Traverses the depot data path and
+    # creates an array from each package
+    # artifact that exists in that path
+
+    TARGETDIR=${1}
+    for f in ${TARGETDIR}/*; do
+        if [[ -d "${f}" ]]; then
+            getHarts "${f}"
+        elif [[ "${f}" == *.hart ]]; then
+            echo "Found: $f"
+            artifacts+=("${f}")
+        else
+            echo "$f is not a hartfile, skipping."
+            echo ""
+        fi
+    done
+}
+
+uploadHarts() {
+    # Takes the artifact array generated via
+    # traversing the data path and uploads to s3
+    if [ ${#artifacts[@]} -eq 0 ]; then
+        echo ""
+        echo "No artifacts found in Artifact storage path!"
+        echo "Exiting!"
+        exit 1
+    fi
+        keyMsg
+    for hart in "${artifacts[@]}"; do
+        uploadMsg
+        keyId=$(generateKey "${hart}")
+        putArtifacts "${hart}" "${S3_BUCKET}" "${AWS_REGION}" "${keyId}"
+    done
+}
```

Comment:
Super minor - would be nice to echo out a final count of packages successfully migrated.